controllers/apply-next/form-router.js

```diff
@@ -33,92 +37,70 @@ function initFormRouter(formModel) {
     router.use(cached.csrfProtection, async (req, res, next) => {
         res.locals.setSessionData = (dataPath, value) => set(req.session, `${sessionKeys.form}.${dataPath}`, value);
         res.locals.getSessionData = dataPath => get(`${sessionKeys.form}.${dataPath}`)(req.session);
-        res.locals.unsetSessionData = dataPath => unset(req.session, `${sessionKeys.form}.${dataPath}`);
         res.locals.clearSession = () => delete req.session[sessionKeys.form];
 
+        const currentEditingId = res.locals.getSessionData(sessionKeys.editingId);
+
+        res.locals.currentlyEditingId = currentEditingId;
+
         // Look up the current application data
-        let applicationData = {};
-        res.locals.currentlyEditingId = res.locals.getSessionData(sessionKeys.editingId);
-        if (res.locals.currentlyEditingId) {
-            applicationData = await ApplicationService.getApplicationById(
-                sessionKeys.form,
-                res.locals.currentlyEditingId
-            );
-        }
-        res.locals.currentApplicationData = applicationData ? applicationData.application_data : false;
+        const application = currentEditingId
+            ? await applicationsService.getApplicationById(sessionKeys.form, currentEditingId)
+            : {};
+
+        const currentApplicationData = get('application_data')(application);
+        res.locals.currentApplicationTitle = get('application_title')(application);
+        res.locals.currentApplicationData = currentApplicationData;
 
         // Translate the form object for each request and populate it with current user input
-        res.locals.form = prepareForm(req.i18n.getLocale(), formModel, res.locals.currentApplicationData);
+        const form = enhanceForm(req.i18n.getLocale(), formModel, currentApplicationData);
 
-        // Share some useful form variables
-        res.locals.validation = res.locals.getSessionData(sessionKeys.validation);
+        res.locals.form = form;
+        res.locals.formTitle = form.title;
+        res.locals.formBaseUrl = req.baseUrl;
         res.locals.FORM_STATES = FORM_STATES;
-        res.locals.user = req.user;
+        res.locals.breadcrumbs = [{ label: form.title, url: req.baseUrl }];
 
-        res.locals.formBaseUrl = req.baseUrl;
-        res.locals.formTitle = 'Application form: ' + res.locals.form.title;
+        res.locals.user = req.user;
         res.locals.isBilingual = formModel.isBilingual;
         res.locals.enablePrompt = false; // Disable prompts on apply pages
         res.locals.bodyClass = 'has-static-header'; // No hero images on apply pages
-        res.locals.breadcrumbs = [{ label: res.locals.form.title, url: req.baseUrl }];
 
         next();
     });
 
     /**
      * Route: Start page
      */
+    async function renderStartPage(req, res, errors = []) {
+        const { startPage } = res.locals.form;
```

Comment:
Any reason not to keep throwing the errors for missing startpages etc?

Reply:
Yep, I think we can better guard against this by validating the form schema with `joi` or via a unit test. We can catch this earlier than at render.